src/include/storage/sql_table.h

```diff
@@ -194,8 +200,31 @@ class SqlTable {
   ProjectionMap ProjectionMapForOids(const std::vector<catalog::col_oid_t> &col_oids);
 
  private:
-  FRIEND_TEST(WriteAheadLoggingTests, AbortRecordTest);
-  FRIEND_TEST(WriteAheadLoggingTests, NoAbortRecordTest);
+  friend class RecoveryManager;  // Needs access to OID and ID mappings
+  friend class terrier::RandomSqlTableTransaction;
+  friend class terrier::LargeSqlTableTestObject;
+  friend class RecoveryTests;
+
+  // TODO(Gus): We should probably cache this somewhere?
+  /**
+   * Get all column oids for this table
+   * @return oids
+   */
+  std::vector<catalog::col_oid_t> GetAllOids() {
```

Comment:
Since we don't bootstrap during recovery, the schemas for catalog tables may not be in pg_class yet when we call `DatabaseCatalog::GetSchema`. I could do what I do with `GetSqlTable` which hardcodes schema tables, but it would require stashing the table schemas (for just catalog tables) in `DatabaseCatalog`. Is that a lot of additional overhead?